### DISCUSSION TITLE
Rendering images rotated by 90° & flipped?

### DISCUSSION THREAD

**Discussion by akavel** [hecrj/iced#1064] [Q&A] [answered]
I'm trying to render a JPEG image. Now, JPEGs have this internal "orientation" Exif data, which when ignored, makes images all rotated in various ways. I found a crate decoding that information, but now my question is: can I ask `widget::Image` (or at least `Primitive::Image`) to use some specific rotation (multiple of 90°) and (potentially) horizontal/vertical flip transformation when rendering? I would prefer that to having to decode jpeg into a memory buffer, then rotate in memory, then encode again to jpeg or something for `Image`...

If there's no way to render flipped/rotated (90deg) images, could I try adding it as a contribution? If yes, where would you suggest I start? I assume something related to backends possibly?

**Comment by hecrj** [comment, marked as answer]
We could use [the `exif` crate](https://docs.rs/kamadak-exif/0.5.4/exif/) to read the orientation. Then, we can apply the transformation when we load the image in `iced_wgpu` in `image::raster::load`:

https://github.com/iced-rs/iced/blob/4d4c61b887e5721d9eff4085052b24722d1aa082/wgpu/src/image/raster.rs#L38

Maybe we can leverage [`image::imageops`](https://docs.rs/image/0.23.14/image/imageops/index.html) to perform the actual rotation.

**Reply by akavel** [reply]
Cool, sounds like an approach that should not be too hard for me to try contributing, thank you so much for the answer!

**Reply by akavel** [reply]
@hecrj I submitted a proposed PR at #1067, would be grateful if you could take a look!
